wurst/lib/Transformation.wurst

```diff
@@ -115,3 +115,12 @@ public function transformUnit(unit origin, int targetID)
 
     // Destroy local resources.
     destroy key
+
+init
+    // Registering a callback to update the unit proper name after transformation
+    registerAfterEffect() (unit target, int unitID) ->
+        // Only hero have proper name
+        if IsHeroUnitId(target.getTypeId())
+            let tmpUnit = createUnit(target.getOwner(), target.getTypeId(), target.getPos(), target.getFacingAngle())
```

Comment:
Creating this at the position of the unit seems like it could have side effects. This will set off triggers for entering a region, won't it? Use the recipe island for this or some other area that is unused for actual gameplay.

Reply:
Actually `boundMax` from `MapBounds` is more sustainable. That's provided by the standard library and is used for the `InstantDummyCaster`. This way we don't have a random reference to the recipe island.